kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/TestUtils.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.utils;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.function.BooleanSupplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.kroxylicious.systemtests.k8s.exception.WaitException;
+
+/**
+ * The type Test utils.
+ */
+public class TestUtils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TestUtils.class);
+
+    /**
+     * Poll the given {@code ready} function every {@code pollIntervalMs} milliseconds until it returns true,
+     * or throw a WaitException if it doesn't return true within {@code timeoutMs} milliseconds.
+     * @param description the description
+     * @param pollIntervalMs the poll interval ms
+     * @param timeoutMs the timeout ms
+     * @param ready the ready
+     * @return The remaining time left until timeout occurs (helpful if you have several calls which need to share a common timeout),
+     */
+    public static long waitFor(String description, long pollIntervalMs, long timeoutMs, BooleanSupplier ready) {
+        return waitFor(description, pollIntervalMs, timeoutMs, ready, () -> {
+        });
+    }
+
+    /**
+     * Wait for long.
+     *
+     * @param description the description
+     * @param pollIntervalMs the poll interval ms
+     * @param timeoutMs the timeout ms
+     * @param ready the ready
+     * @param onTimeout the on timeout
+     * @return the long
+     */
+    public static long waitFor(String description, long pollIntervalMs, long timeoutMs, BooleanSupplier ready, Runnable onTimeout) {
+        LOGGER.debug("Waiting for {}", description);
+        long deadline = System.currentTimeMillis() + timeoutMs;
+        String exceptionMessage = null;
+        int exceptionCount = 0;
+        StringWriter stackTraceError = new StringWriter();
+
+        while (true) {
+            boolean result;
+            try {
+                result = ready.getAsBoolean();
+            }
+            catch (Exception e) {
+                exceptionMessage = e.getMessage();
+                if (++exceptionCount == 1 && exceptionMessage != null) {
+                    // Log the first exception as soon as it occurs
+                    LOGGER.error("Exception waiting for {}, {}", description, exceptionMessage);
```

Comment:
Given we only log this once I think we should just log the exception here.
```suggestion
                    LOGGER.error("Exception waiting for {}, {}", description, exceptionMessage, e);
```